static/js/public/preview.js

```diff
@@ -0,0 +1,300 @@
+import Remarkable from "remarkable";
+import { default as initScreenshots } from "./snap-details/screenshots";
+
+// Ensure markdown is set to be the same as `webapp/markdown.py` config
+// doesn't include the custom ascii bullet-point (as this is legacy
+// and shouldn't be promoted).
+const md = new Remarkable({
+  linkify: true
+});
+md.core.ruler.disable([
+  "references",
+  "footnote_tail",
+  "abbr2",
+  "replacements",
+  "smartquotes"
+]);
+md.block.ruler.disable([
+  "blockquote",
+  "hr",
+  "footnote",
+  "heading",
+  "lheading",
+  "htmlblock",
+  "table"
+]);
+
+// For the different elements we might need to change different properties
+const functionMap = {
+  title: "innerHTML",
+  summary: "innerHTML",
+  description: "innerHTML",
+  website: "href",
+  contact: "href",
+  screenshots: "appendChild",
```

Comment:
Screenshots seem to be treated separately anyway (as a function rather then property), so why not getting it out of here and just handling where they are handled (without functionMap)?

Reply:
Another approach (thay may need a bit more work but may be more flexible and scallable for future changes) would be to create "functionMap" as something like:

```js
const functionMap = {
   title: (el, content) => el.innerHTML = content,
   website: (el, content) => el.href = content,
   //...
}
```

This way screenshots may have their own dedicated (more complex) function.

But I'm not sure how much it would actually simplify things from current state.

Reply:
Both good comments, I started the functionMap before getting to the complexity of screenshots, so well worth a refactor to neaten things up.